server/service/transport.go

```diff
@@ -217,7 +217,7 @@ func hostListOptionsFromRequest(r *http.Request) (fleet.HostListOptions, error)
 
 	teamID := r.URL.Query().Get("team_id")
 	if teamID != "" {
-		id, err := strconv.Atoi(teamID)
+		id, err := strconv.ParseUint(teamID, 10, 32)
```

Comment:
This is for better validation, `Atoi` allows signed integers while `ParseUint` doesn't.